jwt-native/src/main/java/org/ballerinalang/stdlib/jwt/JwksClient.java

```diff
@@ -0,0 +1,162 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.jwt;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * Extern function to call JWKs endpoint using the JDK11 HttpClient and return the payload of the HTTP response.
+ */
+public class JwksClient {
+
+    public static Object getJwksResponse(BString url, BMap<BString, Object> clientConfig) {
+        String httpVersion = clientConfig.getStringValue(StringUtils.fromString(Constants.HTTP_VERSION)).getValue();
+        BMap<BString, Object> secureSocket =
+                (BMap<BString, Object>) getMapValueIfPresent(clientConfig, Constants.SECURE_SOCKET);
+        if (secureSocket != null) {
+            boolean disable = secureSocket.getBooleanValue(StringUtils.fromString(Constants.DISABLE));
+            if (disable) {
+                try {
+                    SSLContext sslContext = initSslContext();
+                    HttpClient client = buildHttpClient(httpVersion, sslContext);
+                    return callJwksEndpoint(client, url.getValue());
+                } catch (NoSuchAlgorithmException | KeyManagementException e) {
+                    return createError("Failed to init SSL context. " + e.getMessage());
+                }
+            }
+            BMap<BString, BString> trustStore =
+                    (BMap<BString, BString>) getMapValueIfPresent(secureSocket, Constants.TRUSTSTORE);
+            if (trustStore != null) {
+                try {
+                    SSLContext sslContext = initSslContext(trustStore);
+                    HttpClient client = buildHttpClient(httpVersion, sslContext);
+                    return callJwksEndpoint(client, url.getValue());
+                } catch (Exception e) {
+                    return createError("Failed to init SSL context with truststore. " + e.getMessage());
+                }
+            }
```

Comment:
What happens if the user gives both disable to true, and a trustore? 
Shouldn't we tell the user not to give both at the same time?